lib-model/lib-model-consume-cli/src/main/java/com/uber/nullaway/libmodel/LibModelInfoExtractor.java

```diff
@@ -0,0 +1,287 @@
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class LibModelInfoExtractor {
+
+  public static void main(String[] args) {
+    Map<String, MethodAnnotationsRecord> methodRecords = new LinkedHashMap<>();
+    LibModelInfoExtractor libModelInfoExtractor = new LibModelInfoExtractor();
+    // input directory
+    libModelInfoExtractor.processDirectory(args[0], methodRecords);
+    // output directory
+    libModelInfoExtractor.writeToAstubx(args[1], methodRecords);
+  }
+
+  public void processDirectory(String file, Map<String, MethodAnnotationsRecord> methodRecords) {
+    Path root = dirnameToPath(file);
+    AnnotationCollectorCallback mc = new AnnotationCollectorCallback(methodRecords);
+    CollectionStrategy strategy = new ParserCollectionStrategy();
+    // Required to include directories that contain a module-info.java, which don't parse by
+    // default.
+    strategy.getParserConfiguration().setLanguageLevel(LanguageLevel.JAVA_17);
+    ProjectRoot projectRoot = strategy.collect(root);
+
+    projectRoot
+        .getSourceRoots()
+        .forEach(
+            sourceRoot -> {
+              try {
+                sourceRoot.parse("", mc);
+              } catch (IOException e) {
+                System.err.println("IOException: " + e);
+              }
+            });
+  }
+
+  public void writeToAstubx(String outputPath, Map<String, MethodAnnotationsRecord> methodRecords) {
+    Map<String, String> importedAnnotations =
+        ImmutableMap.<String, String>builder()
+            .put("Nonnull", "javax.annotation.Nonnull")
+            .put("Nullable", "javax.annotation.Nullable")
+            .build();
+    DataOutputStream dos;
+    try {
+      Path opPath = Paths.get(outputPath);
+      Files.createDirectories(opPath.getParent());
+      dos = new DataOutputStream(Files.newOutputStream(opPath));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    if (!methodRecords.isEmpty()) {
+      try {
+        StubxWriter.write(
+            dos, importedAnnotations, new HashMap<>(), new HashMap<>(), methodRecords);
+        dos.close();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  public Path dirnameToPath(String dir) {
+    File f = new File(dir);
+    if (!f.exists()) {
+      System.err.printf("Directory %s (%s) does not exist.%n", dir, f);
+      System.exit(1);
+    }
+    if (!f.isDirectory()) {
+      System.err.printf("Not a directory: %s (%s).%n", dir, f);
+      System.exit(1);
+    }
+    String absoluteDir = f.getAbsolutePath();
+    if (absoluteDir.endsWith("/.")) {
+      absoluteDir = absoluteDir.substring(0, absoluteDir.length() - 2);
+    }
```

Comment:
I don't think we need all this error checking for now.  It's ok to just skip it, we'll get an exception if something wrong occurs I think